src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -89,9 +89,11 @@ class Crystal::Doc::Generator
 
     main_index = Main.new(raw_body, Type.new(self, @program), project_info)
     File.write File.join(@output_dir, "index.json"), main_index
+    File.write File.join(@output_dir, "search-index.json"), main_index.to_json_search
     File.write File.join(@output_dir, "search-index.js"), main_index.to_jsonp
 
     File.write File.join(@output_dir, "404.html"), MainTemplate.new(Error404Template.new.to_s, types, project_info)
+    File.write File.join(@output_dir, "schema.json"), SchemaTemplate.new
```

Comment:
The idea was that the search-index format shouldn't really be depended upon since it's aggressively reduced to only contain what is needed for search, whereas the full schema is more robust. Although there honestly isn't a good reason to include said schema in the output at all, I'm open to changing it for what makes more sense